### PR TITLE
Fix R 3.3.0-3.3.2 builds

### LINKED ISSUE #178: Some builds are missing for Debian 12
Namely: 3.3.2, 3.3.1, 3.3.0. On the arm64 builds I see this:
```
debian-12_1     | checking for zlib.h... yes
debian-12_1     | checking if zlib version >= 1.2.5... no
debian-12_1     | checking whether zlib support suffices... configure: error: zlib library and headers are required
builder_debian-12_1 exited with code 1
```

### PR DESCRIPTION
**Pull request by gaborcsardi**
Closes #178.

### CODE REVIEW

File: builder/Dockerfile.centos-7 [hunk 1]
```diff
@@ -73,4 +73,5 @@ ENV R_RD4PDF="times,hyper"
 
 COPY package.centos-7 /package.sh
 COPY build.sh .
+COPY patches /patches
```
Comment (glin) [inline builder/Dockerfile.centos-7:76]: Based on previous notes, CentOS 7 was supposedly unaffected, and I do see an installer up at https://cdn.rstudio.com/r/centos-7/pkgs/R-3.3.0-1-1.x86_64.rpm

**Review by glin — COMMENTED** [review]
Nice! We've had this around forever and can finally remove the caveat note from the docs.

Just one more thing, handler.py will need to be updated to not skip these builds anymore: https://github.com/rstudio/r-builds/blob/b541a8ff7cb0d8a459338520cc5bec99a2076048/handler.py#L149-L154

This should also close https://github.com/rstudio/r-builds/issues/76 and https://github.com/rstudio/r-builds/issues/36

File: builder/Dockerfile.centos-7 (hunk 1 above)
Comment (gaborcsardi) [reply builder/Dockerfile.centos-7:76]: Oh, OK. OTOH, it also does not hurt to do it for CentOS 7 as well, and then it is simpler if we do it everywhere.

**Comment by gaborcsardi** [comment]
OK, I updated `handler.py` hopefully it is OK now.

Let me know if you want to remove the patch from CentOS 7. I think it is better to keep it, one less special case. I did try the build with the patch on CentOS 7 (in the arm64 builds) and it works fine.

**Review by glin — APPROVED** [review]

File: builder/Dockerfile.centos-7 (hunk 1 above)
Comment (glin) [reply builder/Dockerfile.centos-7:76]: Sure, that's a fine reason to keep it in then.

**Pull request merged**